### ISSUE TITLE
Game Config

### ISSUE THREAD

**Issue by Phoscur** [ugml/api#85]
Trying to get back to developing the client ;)

Proposing some changes/simplifications for the interfaces:
- always group resources in a separate object (ResourceCollection)
- what are these mappings provided for? remove them, we can generate those easily if needed
- inline game config properties into main-config with subsidiary units config
- return all config at once in a single /config query

Typed the Game and Units Configuration:
```typescript


export interface Configuration {
  game: GameConfiguration;
}

export interface GameConfiguration {
  speed: number;
  start: {
    resources: ResourceCollection;
    planet: {
      name: string;
      diameter: number;
      fields: number;
      resources: ResourceCollection;
      position: {
        galaxy: number;
        system: number;
        planet: number;
      };
    };
  };
  units: UnitConfiguration;
}

export interface ResourceCollection {
  metal: number;
  crystal: number;
  deuterium: number;
}

export interface Costs extends ResourceCollection {
  energy: number;
  factor: number;
}

export interface UnitConfiguration {
  buildings: Building[];
  ships: Unit[];
  requirements: Requirements;
}

export type ID = number;

export interface Unit {
  unitID: ID;
  costs: Costs;
}

export interface ShipProperties {
  consumption: number;
  speed: number;
  capacity: number;
}

export interface RapidFireAgainst {
  unitID: ID;
  amount: number;
}

export interface Building extends Unit {}
export interface Ship extends Unit {
  stats: ShipProperties;
  rapidfire: RapidFireAgainst[];
}
export interface Defense extends Unit {}
export interface Technology extends Unit {}

```

**Comment by Phoscur** [comment]
@mamen no comment?

**Comment by mamen** [comment]
Looks good to me. Only the position-property for the start-planet can be removed as it is randomly generated.

**Comment by Phoscur** [comment]
> Looks good to me. Only the position-property for the start-planet can be removed
Maybe we rename it to positions and actually save maximum values in there?

**Comment by mamen** [comment]
That would make more sense, but why would the client need that information? Or do you mean min/max for the size of the galaxies and solar-systems?

**Comment by Phoscur** [comment]
Well yeah make it min, max e.g. ```
positions: {
        galaxy: number[]; // [1,9]
        system: { // or
           min: number; // 1
           max: number; // 99
        },
        planet: number[];  // [1,15]
```

This makes me think of position based partitioning... well the array approach is more compact and might be more flexible to support a different configuration (3+ elements, seperate values, could be strings too?).

**Comment by mamen** [comment]
@Phoscur have a look at the new PR I've just linked. It should implement what you've proposed

**Comment by mamen** [comment]
Merged with #89
